internal/sops/awskms/keysource.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awskms
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"encoding/base64"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	arnRegex        = `^arn:aws[\w-]*:kms:(.+):[0-9]+:(key|alias)/.+$`
+	stsSessionRegex = "[^a-zA-Z0-9=,.@-]+"
+	// kmsTTL is the duration after which a MasterKey requires rotation.
+	kmsTTL = time.Hour * 24 * 30 * 6
+)
+
+// MasterKey is a AWS KMS key used to encrypt and decrypt sops' data key.
+type MasterKey struct {
+	Arn               string
+	Role              string
+	EncryptedKey      string
+	CreationDate      time.Time
+	EncryptionContext map[string]string
+
+	credentialsProvider aws.CredentialsProvider
+
+	// epResolver IS ONLY MEANT TO BE USED FOR TESTS.
+	// it can be used to override the endpoint that the AWS client resolves to
+	// by default. it's hacky but there is no other choice, since you can't
+	// specify the endpoint as an env var like you can do with an access key.
+	epResolver aws.EndpointResolver
+}
+
+// CredsProvider is a wrapper around aws.CredentialsProvider used for authenticating
+// when using AWS KMS.
+type CredsProvider struct {
+	credsProvider aws.CredentialsProvider
+}
+
+// NewCredsProvider returns a Creds object with the provided aws.CredentialsProvider
+func NewCredsProvider(cp aws.CredentialsProvider) *CredsProvider {
+	return &CredsProvider{
+		credsProvider: cp,
+	}
+}
+
+// ApplyToMasterKey configures the credentials the provided key.
+func (c CredsProvider) ApplyToMasterKey(key *MasterKey) {
+	key.credentialsProvider = c.credsProvider
+}
+
+// LoadAwsKmsCredsProviderFromYaml parses the given yaml returns a CredsProvider object
+// which contains the credentials provider used for authenticating towards AWS KMS.
+func LoadAwsKmsCredsProviderFromYaml(b []byte) (*CredsProvider, error) {
```

Comment:
```suggestion
// LoadCredsProviderFromYAML parses the given YAML returns a CredsProvider object
// which contains the credentials provider used for authenticating towards AWS KMS.
func LoadCredsProviderFromYaml(b []byte) (*CredsProvider, error) {
```

(the package name is already context defining).